crates/red_knot_python_semantic/src/types.rs

```diff
@@ -994,16 +994,12 @@ impl<'db> ClassType<'db> {
 pub struct UnionType<'db> {
     /// The union type includes values in any of these types.
     #[return_ref]
-    elements: FxOrderSet<Type<'db>>,
+    elements: Box<[Type<'db>]>,
```

Comment:
I saw that it's now necessary to do some awkward dereferencing because the salsa accessor returns `&Box<[Type<'db>]`. What I liked to do in the past is to implement my manual accessor:

```rust
    #[return_ref]
	  elements_slice: Box<[Type<'db>]>

    ...


impl <'db> UnionType<'db> [
	fn elements(&self, db: &'db dyn Db) -> &'db [Type<'db>] {
		&*self.elements_slice(db)
	}
```

Reply:
> It would be nice if we could use `SmallVec` here.... except that we can't because of its lifetime variance bug

I feel like `SmallVec` isn't that complicated, and maybe we should just implement our own version of it.

Reply:
An as efficient version as small vec is a bit involved. But let's do this as a separate PR

Reply:
Implemented the accessor to simplify usage of union elements; thanks for the suggestion!